File: SocketTCP/java/src/bai1/DateClient.java
package bai1;
import java.io.*;
import java.net.*;

public class DateClient {

	public static void main(String[] args) {
		try {
            Socket client = new Socket("localhost", 5000);
            
            BufferedReader in = new BufferedReader(new InputStreamReader(client.getInputStream()));
            
            String time = in.readLine();
            System.out.println("Ngày tháng năm từ Server: " + time);
            
            client.close();
        } catch (IOException e) {
            System.err.println(e);
        }

	}

}

File: SocketTCP/dotnet/bai1/DateClient.cs
using System;
using System.Net;
using System.Net.Sockets;
using System.Text;
using System.IO;

namespace bai1
{
    public class DateClient
    {
        public static void Run()
        {
            const string serverIp = "localhost";
            const int port = 5001;
            try
            {
                using (TcpClient client = new TcpClient())
                {
                    client.Connect(serverIp, port);
                    using (NetworkStream ns = client.GetStream())
                    using (StreamReader sr = new StreamReader(ns, Encoding.UTF8))
                    {
                        string? time = sr.ReadLine();
                        Console.WriteLine("Ngày tháng năm từ Server: " + (time ?? "(no data)"));
                    }
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine("Error: " + ex.Message);
            }
        }
    }
}